### DISCUSSION TITLE
My fork has updated to debian12

### DISCUSSION THREAD

**Discussion by CicadaSeventeen** [SpiralLinux/SpiralLinux-project#264] [General]
Hi,
Long time no see. Debian12 (Bookworm) is about to be launched, and it's rc version is available now. Do you remember once I forked SpiralLinux? Now I have updated it to Bookworm.
However, it is not only an update but I also make some changes For example, I remove many software because I like to use flatpak. But you can still undo some of my changes to easily update SpiralLinux to Bookworm.

**Comment by CicadaSeventeen** [comment]
see https://github.com/DebianEasy/SpiralLinux-fork-bookworm

**Comment by CicadaSeventeen** [comment]
@geckolinux I find my fork cannot install gurb. Could you help to think about why?

**Reply by geckolinux** [reply]
Hi there @CicadaSeventeen , good to see you. 

> I find my fork cannot install gurb

Do you mean that it can't build the ISO with the GRUB packages, or it can't install GRUB to the disk with Calamares?

**Reply by CicadaSeventeen** [reply]
> Hi there @CicadaSeventeen , good to see you.
> 
> > I find my fork cannot install gurb
> 
> Do you mean that it can't build the ISO with the GRUB packages, or it can't install GRUB to the disk with Calamares?

I can build iso, but `calamares` fails to install it.
It turns out that caused by `grub-btrfs` 4.12. On the other hand, `grub-btrfs` 4.11 does not have such problem.
My fork updates `grub-btrfs` 4.12 so this happen. Make sure that please do not update it if you still want to use `grub-btrfs` on SpiralLinux

**Reply by CicadaSeventeen** [reply]
BTW, `grub-btrfs` can be found in kalilinux's repo, and it works on debian well

**Reply by geckolinux** [reply]
Maybe try this: In the initial `grub-btrfs` `config` file it needs: `GRUB_BTRFS_DISABLE="true"`

**Reply by fweller** [reply]
Hi @CicadaSeventeen, 

Very cool effort with your Debian Easy ISO.  I gave the KDE Plasma spin a try in VirtualBox.  There are a few rough edges like not having a browser but a blank icon (probably originally for firefox) is on the menu bar.  I think you are about to find out how challenging it is to build an ISO for general consumption, but I encourage you to make this your own distro and customize it as you like.  

I am extremely grateful to @geckolinux for starting the process of seeding the internet with a base to do this.  I hope to see many efforts such as your own to make Debian the distro to customize and share.

**Comment by GREG3f** [comment]
I was unable to boot the Spiral Linux KDE Distro as it just hung on boot so I tried the EasyDebian ISO and was able to get the OS installed and I would like to say I really like you distro!  Thank you for all your efforts  @geckolinux and @CicadaSeventeen !!!

Unfortunately the only feature that I am unable to get figured out is Snapper and Grub.  I am able to change a couple of Grub Settings with the Grub Customizer Tool but there are no entries for any of my restore points and I have not been able to change the Appearance settings at all. I can live with the Grub Appearance as is but would love to figure out a way to get Snapper images available to restore at boot. Any suggestions appreciated!

**Reply by CicadaSeventeen** [reply]
I remove `grub-btrfs` for it can break installation if I pre-install it. You may find that I add a repository into DebianEasy, you can just use `apt install grub-btrfs` to install it.
After Installation You can choose snapper image when booting.

**Comment by GREG3f** [comment]
That did the trick! Thank you @CicadaSeventeen !

**Comment by methanoid** [comment]
Hi

Been playing with your DebianEasy and whilst I like the "de Spiraling" you've done (I hate the Zwopper green pic myself) I find some of your decisions puzzling

You've actively REVERTED in many places the "--no-install-recommends" that means you get a much bigger install and ISO. I wanted to slim Spiral down, not Bloat it up!! So I've been "fixing" those and the many typos. Your build files include 1Gb of firmware files which needn't be downloaded. 

FYI, your ISO for KDE Plasma is 2.8GB, mine is 1.8GB and that's with me adding some bits in including:

bleachbit, breeze, cups-bsd, goverlay, libavcodec-extra, make, mangohud, mc, ncdu, preload, python3, python3-pip, rclone, ufw, steam-devices, mpv, elisa  (some libs I dont know WHY I added so will remove!) plus I added back a few bits you removed which I saw no reason to cut out.

**Reply by fweller** [reply]
I love the vibe here.  @geckolinux posts a project to improve and customize the Debian install process, @CicadaSeventeen takes the project and makes his own flavor, which @methanoid picks up to make his own flavor.  This is how open source works.  It would be awesome if one of you would write up a step-by-step list of instructions on how you forked SpiralLinux, how you customized it for your needs, and then how you built it.   This would enable a lot more people to make their own customization projects.

**Reply by geckolinux** [reply]
@fweller 
> It would be awesome if one of you would write up a step-by-step list of instructions on how you forked SpiralLinux, how you customized it for your needs, and then how you built it. 

This is similar information:
https://github.com/orgs/SpiralLinux/discussions/228#discussioncomment-4809266

**Reply by methanoid** [reply]
And thats basically sort of what I did... I'd had some issues trying to take Spiral 11 and make a Spiral 12 so took Cicada's "Spiral 12" and tweaked a few bits but to be honest I found too much I thought was "wrong" to keep this long term. I like the "de-Spiral" done to it (sorry @geckolinux I'm not a fan of your Zwopper Green everywhere - I like stock Debian - and I don't want a Spiral branding - just Debian for me). For me Spiral is close to "right" apart from being themed.  For me, Spiral is just "Debian done right" or "Debian but better" so I'd like it still to show Debian on my machine.

I really like how easy it is to "mod" and I don't want to distribute something that is as obviously "derivative" of Spiral - feels a bit disrespectful to me (but that's just how I feel, nobody else needs to feel like that!).

I just got frustrated and bored waiting for Spiral 12.  I'm hoping @geckolinux manages to squeak out a Spiral 12 soon

Oh and while I am dreaming... I'd love Spiral (or "Debian Done Better") to be ONE ISO which just asks you at install if you want a particular DE....  I wouldnt care if it's a Live DVD of Gnome or Plasma, but one that allows ALL DEs to be installed (with tweaks)

Might be easier to maintain too? 

Anyway, again thanks to @geckolinux - I keep coming back to Spiral for the fun it gives me (and education in Debian & Linux)

**Reply by thesimonsaysthis** [reply]
@methanoid I'm not as tech savvy as you guys but is there somewhere I can try out your ISO? I would love Spiral minus the branding and the NVIDIA drivers. It's subjective but that green wallpaper is one of the worst I can think off. Ok maybe the visual preferences of MX Linux still top that.

**Reply by geckolinux** [reply]
Just a small suggestion: Why not simply change the wallpaper on your SpiralLinux installation? ;-)

**Reply by thesimonsaysthis** [reply]
@geckolinux yes, for sure. We're talking trivialities here. That would never be the reason for me to not use a distro. If there was SpiralLinux Bookworm I'd just use that and remove NVIDIA drivers, change wallpaper etc. But there isn't so I'm interesting in those ISOs  EDIT: we'll if Spiral boots on my hardware I can just upgrade. Problem solved

**Comment by beardogg0524** [comment]
I managed to fork SpiralLinux's XFCE spin into two different ISOs using Debian Sid: Zoonity Desktop and Zoonity Cloud. Both have been performing very nicely with quite a few downloads for each edition.

The main change is using Plasma Discovery in place of GNOME Software (seems much faster to me) as well as add the Vala global menu to the task bar and the Plank dock/launcher.

If anybody is interested, check out my Twitter at @zoonitydesktop

**Reply by Anonymo** [reply]
Do you have it on Github or an alternate link?  Twitter wants me to create an account.

**Reply by beardogg0524** [reply]
No, but I have a link on Mediafire.

Zoonity Desktop - https://www.mediafire.com/file_premium/op2wcuq9nq3vxjq/zoonitydesktop-rolling-amd64.iso/file
Zoonity Cloud - https://www.mediafire.com/file_premium/xi08yhprczzkv0s/zoonitycloud-rolling-amd64.iso/file
